python/ray/tests/test_memory_scheduling.py

```diff
@@ -18,118 +16,41 @@ def ping(self):
-            )
-            _register_all()
-            self.assertRaisesRegexp(
-                ray.tune.error.TuneError,
-                ".*Insufficient cluster resources.*",
-                lambda: tune.run(
-                    "PG",
-                    stop={"timesteps_total": 10000},
-                    config={
-                        "env": "CartPole-v0",
-                        # too large
-                        "object_store_memory": 10000 * 1024 * 1024,
-                        "framework": "tf",
-                    }))
-        finally:
-            ray.shutdown()
-
-    def testTuneWorkerStoreLimit(self):
-        os.environ["TUNE_PLACEMENT_GROUP_AUTO_DISABLED"] = "1"
-
-        try:
-            ray.init(
-                num_cpus=4,
-                _memory=100 * MB,
-                object_store_memory=100 * MB,
-            )
-            _register_all()
-            self.assertRaisesRegexp(
-                ray.tune.error.TuneError,
-                ".*Insufficient cluster resources.*",
-                lambda:
-                tune.run("PG", stop={"timesteps_total": 0}, config={
-                    "env": "CartPole-v0",
-                    "num_workers": 1,
-                    # too large
-                    "object_store_memory_per_worker": 10000 * 1024 * 1024,
-                    "framework": "tf",
-                }))
-        finally:
-            ray.shutdown()
-
-    def testTuneObjectLimitApplied(self):
-        try:
-            ray.init(num_cpus=2, object_store_memory=500 * MB)
-            result = tune.run(
-                train_oom,
-                resources_per_trial={"object_store_memory": 150 * 1024 * 1024},
-                raise_on_failed_trial=False)
-            self.assertTrue(result.trials[0].status, "ERROR")
-            self.assertTrue("ObjectStoreFullError: Failed to put" in
-                            result.trials[0].error_msg)
-        finally:
-            ray.shutdown()
+def test_memory_request():
+    try:
+        ray.init(num_cpus=1, _memory=200 * MB)
+        # fits first 2
+        a = Actor.remote()
+        b = Actor.remote()
+        ok, _ = ray.wait(
+            [a.ping.remote(), b.ping.remote()], timeout=60.0, num_returns=2)
+        assert len(ok) == 2
+        # does not fit
+        c = Actor.remote()
+        ok, _ = ray.wait([c.ping.remote()], timeout=5.0)
+        assert len(ok) == 0
+    finally:
+        ray.shutdown()
+
+
+def test_object_store_memory_reporting():
+    try:
+        ray.init(num_cpus=1, object_store_memory=500 * MB)
+        time.sleep(0.5)
+        assert ray.available_resources()["object_store_memory"] == 10.0
+        x1 = ray.put(np.zeros(150 * 1024 * 1024, dtype=np.uint8))
+        time.sleep(0.5)
+        assert ray.available_resources()["object_store_memory"] == 7.0
+        x2 = ray.put(np.zeros(75 * 1024 * 1024, dtype=np.uint8))
+        time.sleep(0.5)
+        assert ray.available_resources()["object_store_memory"] == 5.5
```

Comment:
Could we also check that a running task's args will count towards the memory usage?